### DISCUSSION TITLE
Filter by hub sites

### DISCUSSION THREAD

**Discussion by MirasMNC** [microsoft-search/pnp-modern-search#1907] [General]
Do you know if exist a way how to filter search results based on hub sites? Currently I have 3 hub sites and search through all of them (not within whole tenant). I would like to use as a one of filters "Hub sites" but I have not found some out of box column which I could use for that. Is that even possible? I could be possible to use custom column but it is not so practical. Thank you for sharing your experience.

**Comment by wobba** [comment]
`DepartmentId:<site id of hub>` or `RelatedHubSites:<site id of hub>`. The latter one will also cover hub hierarchies. 

I will add public Microsoft Search documentation on these properties as well when I have time.

See:
* https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#connections-tokens
* https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#site-web-hub-etc-tokens
* https://www.techmikael.com/2018/04/working-with-hub-sites-and-search-api.html

**Comment by MirasMNC** [comment]
I am sorry for late response. I have read the documentation and I understand how to use tokens within Search Results web part but I am not sure how to use it in my case Search Filters. The case is that I have a lot of results across tenant. I would like to provide the option for end users that they can use Search Filter "Hub site" to limit scope Search Results only for this hub site. I know that I can use the column "SiteTitle" to limit results only for specific site. I just do not know if is it possible to do similar way for hub sites.
